lib/remote_console/web_mks.rb

```diff
@@ -0,0 +1,41 @@
+module RemoteConsole
+  class WebMKS < SSLSocket
+    attr_accessor :url
+
+    def initialize(record, socket)
+      super(record, socket)
+      @url = URI::Generic.build(:scheme => 'wss',
+                                :host   => @record.host_name,
+                                :port   => @record.port,
+                                :path   => path).to_s
+      @driver = WebSocket::Driver.client(self, :protocols => [protocol])
+      @driver.on(:close) { socket.close unless socket.closed? }
+      @driver.start
+    end
+
+    def fetch(length)
+      @driver.on(:message) { |msg| yield(msg.data) } if @driver.listeners(:message).length.zero?
+
+      data = @ssl.sysread(length)
+      @driver.parse(data)
+    end
+
+    def issue(data)
+      @driver.binary(data)
+    end
+
+    def write(data)
```

Comment:
could you move this to `SSLSocket`

and change `issue` in parent to:

```ruby
  def issue(data)
    write(data)
  end
```
or alias it or what ever

Reply:
No, the [websocket-driver](https://github.com/faye/websocket-driver-ruby) requires the `write` method, so I intentionally created the `fetch` and `issue` methods to create a layer that abstracts away the differences between a WebSocket and a Regular/SSLSocket connection. The `write` here is being called internally by the driver only.